rollup/sync_service/sync_service.go

```diff
@@ -82,7 +84,8 @@ func NewSyncService(ctx context.Context, genesisConfig *params.ChainConfig, node
 	// otherwise there's no way for the node to know if it missed any messages of the V2 queue (as it was not querying it before)
 	// but continued to query the V1 queue (which after V2 deployment does not contain any messages anymore).
 	// this is a one-time operation and will not be repeated on subsequent restarts.
-	if genesisConfig.Scroll.L1Config.L1MessageQueueV2DeploymentBlock > 0 &&
+	if genesisConfig.IsEuclidV2(uint64(time.Now().Unix())) &&
```

Comment:
I think this is wrong, Euclid on-chain might happen a few hours earlier than in l2geth

Reply:
If we're ok with this then I can remove the check